src/actions/ui.js

```diff
@@ -1,12 +1,18 @@
 // @flow
 const constants = require("../constants");
-const { getSource } = require("../selectors");
+const { getSource, getFileSearchState } = require("../selectors");
 import type { ThunkArgs } from "./types";
 
 function toggleFileSearch(searchOn: boolean) {
-  return {
-    type: constants.TOGGLE_FILE_SEARCH,
-    searchOn
+  return ({ dispatch, getState }: ThunkArgs) => {
+    if (searchOn) {
```

Comment:
Hmm, I like that. lets have two actions `toggle` and `close`? 

Also, i suppose the close button will always close given it only is shown when open :)